unit/src/CommandLine.C

```diff
@@ -0,0 +1,73 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "gtest_include.h"
+
+#include "CommandLine.h"
+#include "InputParameters.h"
+
+TEST(CommandLine, parse)
+{
+  {
+    InputParameters params = emptyInputParameters();
+
+    params.addCommandLineParam<bool>(
+        "bool_param_with_default", "--bool_param_with_default", true, "Doc");
+    params.addCommandLineParam<int>(
+        "int_param_with_default", "--int_param_with_default", 42, "Doc");
+    params.addCommandLineParam<std::string>(
+        "string_param_with_default", "--string_param_with_default", "stuff", "Doc");
+
+    params.addCommandLineParam<bool>(
+        "bool_param_without_default", "--bool_param_without_default", "Doc");
+    params.addCommandLineParam<int>(
+        "int_param_without_default", "--int_param_without_default", "Doc");
+    params.addCommandLineParam<std::string>(
+        "string_param_without_default", "--string_param_without_default", "Doc");
+
+    params.addCommandLineParam<std::string>(
+        "compound_string_param", "--compound_string_param", "Doc");
+
+    params.addCommandLineParam<std::string>(
+        "compound_string_param_with_equals", "--compound_string_param_with_equals", "Doc");
+
+    int argc = 9;
+    const char * argv[9] = {"--bool_param_without_default",
```

Comment:
Magic numbers are unnecessary here. You can just initialize the array and pull out the number of entries so we don't have to keep them in sync. This is a unit test so it's no big deal, just a suggestion for better code. Also, the c-array could be a std::array.